File: sdk/go/service/exception_record_service.go
package service

import (
	"log"
	"time"

	"github.com/coffersTech/easy-recon/sdk/go/entity"
	"github.com/coffersTech/easy-recon/sdk/go/repository"
)

// ExceptionRecordService 异常记录服务
type ExceptionRecordService struct {
	reconRepository repository.ReconRepository
}

// NewExceptionRecordService 创建异常记录服务
func NewExceptionRecordService(reconRepository repository.ReconRepository) *ExceptionRecordService {
	return &ExceptionRecordService{
		reconRepository: reconRepository,
	}
}

// RecordReconException 记录对账异常
func (s *ExceptionRecordService) RecordReconException(orderNo, merchantId, msg string, step int) {
	exception := &entity.ReconException{
		OrderNo:       orderNo,
		MerchantId:    merchantId,
		ExceptionType: 1, // 假设 1 为业务异常
		ExceptionMsg:  msg,
		ExceptionStep: step,
		CreateTime:    time.Now(),
		UpdateTime:    time.Now(),
	}

	_, err := s.reconRepository.SaveException(exception)
	if err != nil {
		log.Printf("Failed to save exception record: %v", err)
	}
}

File: sdk/python/service/exception_record_service.py
import logging
from typing import Optional
from repository.recon_repository import ReconRepository
from entity.recon_exception import ReconException

logger = logging.getLogger(__name__)

class ExceptionRecordService:
    """
    Service for recording reconciliation exceptions
    """
    def __init__(self, recon_repository: ReconRepository):
        self.recon_repository = recon_repository

    def record_recon_exception(self, order_no: str, merchant_id: Optional[str], msg: str, step: int):
        try:
            exception = ReconException(
                order_no=order_no,
                merchant_id=merchant_id,
                exception_msg=msg,
                exception_step=step
            )
            self.recon_repository.save_exception(exception)
        except Exception as e:
            logger.error(f"Failed to record exception: {e}")
